go/plugins/firebase/auth.go

```diff
@@ -39,13 +40,12 @@ type AuthClient interface {
 }
 
 // ContextProvider creates a Firebase context provider for Genkit actions.
-func ContextProvider(ctx context.Context, policy AuthPolicy) (core.ContextProvider, error) {
-	app, err := App(ctx)
-	if err != nil {
-		return nil, err
+func ContextProvider(ctx context.Context, g *genkit.Genkit, policy AuthPolicy) (core.ContextProvider, error) {
+	f := genkit.LookupPlugin(g, provider).(*Firebase)
```

Comment:
We should also check if the type assertion is correct so `f, ok := ...` `if !ok { ...`